File: apple/Sources/ReactNativeWatchOS/Modifiers/IgnoresSafeArea.swift
import SwiftUI

/// SwiftUI `.ignoresSafeArea(_:edges:)`. Extends the view past the safe area
/// on the given `edges` (parsed by `RNWEdgeParser`, default `.all`), using
/// the default `.all` safe-area regions.
enum RNWIgnoresSafeAreaModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("ignoresSafeArea") { view, params, _ in
            // `RNWEdgeParser.edgeSet` already defaults a missing/nil value
            // to `.all`, matching the JS default.
            let edges = RNWEdgeParser.edgeSet(params["edges"])
            return AnyView(view.ignoresSafeArea(edges: edges))
        }
    }
}

File: src/modifiers/ignoresSafeArea.ts
import { createModifier } from './createModifier';

export type SafeAreaEdge =
  | 'top'
  | 'bottom'
  | 'leading'
  | 'trailing'
  | 'horizontal'
  | 'vertical'
  | 'all';

export interface IgnoresSafeAreaParams {
  /// Edges along which to extend beyond the safe area. Accepts a single
  /// edge name, `'horizontal'` / `'vertical'` / `'all'`, or an array of
  /// edge names. Defaults to `'all'`.
  edges?: SafeAreaEdge | SafeAreaEdge[];
}

/// SwiftUI `.ignoresSafeArea(_:edges:)`. Lets the view draw into the safe
/// area on the chosen edges (e.g. under the watch bezel insets).
export function ignoresSafeArea(params: IgnoresSafeAreaParams = {}) {
  return createModifier('ignoresSafeArea', params);
}
